File: 2597/2597.ts
function beautifulSubsets(nums: number[], k: number): number {
    function dfs(nums: number[], k: number, i: number, s: Set<number>){
        if (i == nums.length){
            if (s.size === 0){return 0;}
            return 1;
        }
        let count = dfs(nums, k, i+1, s);
        if (s.has(nums[i]-k)){return count;}
        let newS: Set<number>  = new Set(s);
        newS.add(nums[i])
        return count + dfs(nums,k,i+1,newS);
    }

    nums.sort((a, b) => a - b);
    return dfs(nums,k, 0, new Set<number>());
};

File: 2597/2597.py
class Solution:
    def beautifulSubsets(self, nums: List[int], k: int) -> int:
        def dfs(nums,k,i,s) -> int:
            if i >= len(nums): return 1 if s else 0
            count = dfs(nums, k, i+1, s)
            if (nums[i]-k) in s: return count
            return count + dfs(nums, k, i+1, (s | {nums[i]}))

        return dfs(sorted(nums),k,0,set())


        
